PyEMD/checks.py

```diff
@@ -67,49 +75,43 @@ def whitenoise_check(data: np.ndarray, method: str='EMD',  test: str='aposterior
     >>> type(significant_imfs)
     <class 'dict'>
     """
+    assert 0 < alpha < 1, "alpha value should be in between (0,1)"
+    assert test == "apriori" or test == "aposteriori", "Invalid test type"
+    assert isinstance(IMFs, np.ndarray), "Invalid Data type, Pass a numpy.ndarray containing IMFs"
+    assert rescaling_imf > 0 and rescaling_imf <= len(IMFs), "Invalid rescaling IMF"
 
-    assert alpha > 0 and alpha < 1, "alpha value should be in between (0,1)"
-    assert method == 'EMD' or method == 'EEMD' or method == 'CEEMDAN', 'Invalid decomposition method'
-    assert test == 'apriori' or test == 'aposteriori', 'Invalid test type'
-    assert isinstance(data, np.ndarray), 'Invalid Data type, Pass a numpy.ndarray'
+    N = len(IMFs[0])
+    output = {}
+    if N == 0:
+        return {}
+    if np.isnan(np.sum(IMFs)) is True:
+        # Return NaN if input has NaN
+        return None
 
-    N = len(data)
-    if method == 'EMD':
-        emd = EMD(**kwargs)
-        IMFs = emd.emd(normalize(data))
-    elif method == 'EEMD':
-        eemd = EEMD(**kwargs)
-        IMFs = eemd.eemd(normalize(data))
-    elif method == 'CEEMDAN':
-        ceemdan = CEEMDAN(**kwargs)
-        IMFs = ceemdan.ceemdan(normalize(data))
+    if test == "apriori":
+        for idx, imf in enumerate(IMFs):
+            T = math.log(mean_period(imf))
+            energy_density = math.log(energy(imf) / N)
```

Comment:
Both values are in logs but names don't indicate that. With `energy_density` that isn't an issue but `T` usually stands for time related (`[s]` units). Maybe rename to `log_T`? And same with `significance_` method as input to be sure that expectations are met.

Same comment for `aposteriori` section.

Reply:
done